File: video-library-core/src/main/java/it/proconsole/library/video/core/repository/FilmRepository.java
package it.proconsole.library.video.core.repository;

import it.proconsole.library.video.core.model.Film;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

public interface FilmRepository extends ByProtocolRepository {
  Logger logger = LoggerFactory.getLogger(FilmRepository.class);

  List<Film> findAll();

  List<Film> saveAll(List<Film> films);

  default void logStart(Protocol protocol, Operation operation) {
    logger.info("{} - {} request received", protocol, operation);
  }

  default void logStart(Protocol protocol, Operation operation, int inputs) {
    logger.info("{} - {} request received with {} values", protocol, operation, inputs);
  }

  default void logEnd(Protocol protocol, Operation operation, int results) {
    logger.info("{} - {} request completed with {} results", protocol, operation, results);
  }
}

File: video-library-react/src/main/resources/video-library/src/repository/FilmRepository.ts
import { Film } from '../model/Film';
import { Protocol } from '../model/Protocol';
import { fetchHttpClient } from '../utils/HttpClient';
import { ByProtocolRepository } from './ByProtocolRepository';

export interface FilmRepository extends ByProtocolRepository {
  get(): Promise<Film[]>;

  update(films: Film[]): Promise<Film[]>;

  insert(films: Film[]): Promise<Film[]>;
}

const httpClient = fetchHttpClient<Film[] | void, Film[]>();

export const filmRepository = (protocol: Protocol): FilmRepository => {
  return {
    protocol(): Protocol {
      return protocol;
    },
    get(): Promise<Film[]> {
      return httpClient.get({url: `/${protocol}/films`});
    },
    update(films: Film[]): Promise<Film[]> {
      return httpClient.post({url: `/${protocol}/films`, body: films});
    },
    insert(films: Film[]): Promise<Film[]> {
      return httpClient.put({url: `/${protocol}/films`, body: films});
    }
  };
};